src/App.tsx

```diff
@@ -15,33 +16,152 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+const ids = new Array<string>(goodsFromServer.length)
+  .fill('')
+  .map(el => el + uuidv4());
+const minLengths = new Array(goodsFromServer.length)
+  .fill('')
+  .map((el, i) => ({ id: el + uuidv4(), option: i + 1 }));
+
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+function getReorderedGoods(
+  goods: string[],
+  minLen: number,
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods].filter(good => good.length >= minLen);
+
+  if (sortType !== SortType.NONE) {
+    visibleGoods.sort((a, b) => {
+      if (sortType === SortType.ALPABET) {
+        return a.localeCompare(b);
+      }
+
+      return a.length - b.length;
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App: FC = () => {
+  const [isStarted, setIsStarted] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [selected, setSelected] = useState(1);
+
+  const goods = getReorderedGoods(
+    goodsFromServer,
+    selected,
+    sortType,
+    isReversed,
+  );
+
+  return (
+    <div className="App box">
+      {!isStarted
+        ? (
+          <button
+            className="button is-primary is-large"
+            type="button"
+            onClick={() => setIsStarted(true)}
+          >
+            Start
+          </button>
+        )
+        : (
+          <div className="App__content">
+            <div className="App__buttons">
+              <button
+                className={classNames(
+                  'button is-primary',
+                  { 'is-light': sortType === SortType.ALPABET },
+                )}
+                type="button"
+                onClick={() => {
+                  setIsReversed(prevIsReverse => (sortType === SortType.ALPABET
+                    ? !prevIsReverse
+                    : false
+                  ));
```

Comment:
```suggestion
                onClick={() => setIsReversed(!prevIsReverse)};
```
Я робив так але не впевнений, що правильно)))